tests/Datasets/Control.php

```diff
@@ -1,17 +1,22 @@
 <?php
 
+namespace r\Tests\Datasets;
+
+//use function r\db;
+
+
 class Control extends Dataset
 {
-    protected function create()
+    public function populate()
+    {
+    }
+    public function truncate()
     {
-        r\dbCreate('Control')->run($this->conn);
-        r\db('Control')->tableCreate('t1')->run($this->conn);
-    }    
-    
-    protected function delete()
+        \r\db($this->db)->table('t1')->delete()->run($this->conn);
+    }
+
+    public function create()
     {
-        r\dbDrop('Control')->run($this->conn);
+        \r\db($this->db)->tableCreate('t1')->run($this->conn);
     }
 }
-
-?>
```

Comment:
This is a part of PSR-2. See [2.2 Files](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#22-files) for more details. In short:

> The closing `?>` tag MUST be omitted from files containing only PHP.

The reasoning is missing, but in general it is considered better to simply drop the trailing `?>` as it can lead to mistakes where whitespace actually comes after the trailing `?>` that is hard to see. This whitepsace can result in strange errors where the whitespace is rendered (and often interfering with `header()` calls) but you can't find out from where. PHP doesn't require a trailing `?>` so the idea is to simply never include it. That way you never run into a problem with phantom whitespace.